docker/horcrux/native.Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM golang:1.21-alpine AS build-env
+FROM golang:1.24-alpine AS build-env
```

Comment:
I took the liberty to bump go to 1.24. If there is a reason you want to stick to 1.21, I can revert.